blog/_posts/2024-08-21-scala-3.5.0-released.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: blog-detail
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.5.0 released!
+---
+
+![Scala 3.5]({{ site.baseurl }}/resources/img/scala-3.5-launch.jpg)
+
+We are happy to announce that after long and hard work by the entire compiler team, and after seven release candidates, the first version of the Scala 3.5 line is officially out!
+
+## New default runner - Scala CLI
+
+[Scala CLI](https://scala-cli.virtuslab.org/) is a popular tool among Scala devs. It allows lightning-fast running, testing, and prototyping of Scala scripts and single-module projects. In 3.5.0, it becomes part of the default Scala distribution. If you install Scala through popular package managers, such as Brew or SDKMAN!, the installed `scala` command allows you to compile, run, test, and even publish your code on Maven Central.  It has out-of-the-box support for `using` directives, toolkits, compilation to JavaScript and native binaries, and other goodies formerly available only if you installed a separate tool.
+
+### Short example
+
+Given the following file named `biggerThan.scala`
+
+```scala
+//> using dep com.lihaoyi::os-lib:0.10.3
+
+@main def run(path: String, size: Int) =
+  os.list(os.Path(path, os.pwd))
+    .filter: p =>
+      os.size(p) > size * 1024
+    .foreach(println)
+```
+
+We can run `scala biggerThan -- ../my-directory 10`. This will download os-lib and its transitive dependencies, compile the source code, and finally run the compiled program, printing all files in `../my-directory` bigger than 10 kB. Subsequent invocations of the command will use cached bytecode, or if the source code changed, trigger incremental compilation.
```

Comment:
```suggestion
We can run `scala biggerThan.scala -- ../my-directory 10`. This will download os-lib and its transitive dependencies, compile the source code, and finally run the compiled program, printing all files in `../my-directory` bigger than 10 kB. Subsequent invocations of the command will use cached bytecode, or if the source code changed, trigger incremental compilation.
```